### PR TITLE
Avoid forEach on NodeList for optimal backward support

### PR DESCRIPTION
**Pull request by darkwing**
@ngokevin made a good point about using `NodeList.prototype.forEach`/`map` which probably wont fly on older Android/iOS.  Stick with the old I guess.

### CODE REVIEW

**Comment by mkungla** [comment]
@darkwing and @ngokevin wouldn't following be a solution? and perhaps add then polyfill into lib for compatibility.
```js
var medias = Array.from(this.querySelectorAll('audio, videos'));
var imgs = Array.from(this.querySelectorAll('img'));
```

**Comment by darkwing** [comment]
@mkungla I submitted a PR with Array.from last week but @dmarcos thought we should use `Array.prototype.slice` instead so we could accommodate older browsers.

**Comment by mkungla** [comment]
makes sense while there would be possibility to add polyfill for Array.from

**Comment by darkwing** [comment]
@mkungla We could consider adding `querySelectorAll` to the `ANode` prototype which would return an array via `Array.prototype.slice.call(NodeList)` ; that way we're always working with an array when we query.

**Comment by mkungla** [comment]
Selecting nodes without the need to traverse the DOM in tedious loops is nice, however @ngokevin and @dmarcos probably have better overview what performance impact querySelectorAll can have in this context for A-Frame.

**Comment by ngokevin** [comment]
@davidwalsh I noticed a bug, it should be `video` not `videos`.

File: src/core/a-assets.js [hunk 1]
```diff
@@ -30,17 +30,18 @@ module.exports = registerElement('a-assets', {
         }
 
         // Wait for <img>s.
-        imgs.forEach(function (img) {
+        for (var i = 0; i < imgs.length; i++) {
```
Comment (ngokevin) [inline src/core/a-assets.js]: A-Frame likes `vars on top`, might have to add that to our linter.

**Comment by ngokevin** [comment]
I'd file an issue for polyfilling NodeList forEach/map.

**Comment by darkwing** [comment]
Issues fixed.

**Comment by ngokevin** [comment]
Thanks!

**Pull request merged**